controller/src/main/java/org/jboss/as/controller/operations/global/EnhancedSyntaxSupport.java

```diff
@@ -51,16 +54,24 @@ static String extractAttributeName(String expression) {
         }
     }
 
-    static ModelNode resolveEnhancedSyntax(final String attributeExpression, final ModelNode model) throws OperationFailedException {
+    static ModelNode resolveEnhancedSyntax(final String attributeExpression, final ModelNode model, AttributeDefinition attributeDefinition) throws OperationFailedException {
         ModelNode result = model;
-        for (String part : attributeExpression.split("\\.")) {
+        String[] parts = attributeExpression.split("\\.");
+        for (int i = 0; i < parts.length; i++) {
```

Comment:
It would not be null. I added a couple asserts though because that's not obvious.